### PR TITLE
E2E tests framework for kapply

### PR DESCRIPTION
**Pull request by phanimarupaka**
*Why*

This PR is to set-up the e2e tests framework for new cli-utils commands. Added tests for the current progress. We should keep adding tests as we develop features. There are missing pieces in apply and status commands which should be addressed and tests should be added.

*How to run*
make verify-kapply-e2e

### CODE REVIEW

File: examples/alphaTestExamples/helloapp.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+[hello]: https://github.com/monopole/hello
+[kind]: https://github.com/kubernetes-sigs/kind
+
+# Demo: hello app
+
+This demo helps you to deploy an example hello app end-to-end using kustomize.
```
Comment (monopole) [inline examples/alphaTestExamples/helloapp.md]: s/kustomize/`kapply`

File: examples/alphaTestExamples/helloapp.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Steps:
+1. Create the resources files.
+3. Spin-up kubernetes cluster on local using [kind].
+4. Deploy the app using kapply and verify the status.
```
Comment (monopole) [inline examples/alphaTestExamples/helloapp.md]: `kapply`

File: examples/alphaTestExamples/helloapp.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    deployment: hello
+  type: LoadBalancer
+  ports:
+  - protocol: TCP
+    port: 8666
+    targetPort: 8080
+EOF
+```
+
+Create a `grouping.yaml` resource. By this, you are defining the grouping of the current directory, `base`. Kustomize uses the unique label in this file to track any future state changes made to this directory. Make sure the label key is `kustomize.config.k8s.io/inventory-id` and give any unique label value and DO NOT change it in future.
```
Comment (monopole) [inline examples/alphaTestExamples/helloapp.md]: Please wrap long text so the lines are under 80 chars for easy edit and review.
It makes no difference to the rendering.

The first `Kustomize` should be `kapply`, and the 2nd should be `cli-utils.sigs.k8s.io/inventory-id`

File: examples/alphaTestExamples/helloapp.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+The following requires installation of [kind].
+
+Delete any existing kind cluster and create a new one. By default the name of the cluster is "kind"
+<!-- @deleteAndCreateKindCluster @testE2EAgainstLatestRelease -->
+```
+kind delete cluster
+kind create cluster
+```
+
+Use the kapply binary in MYGOBIN to apply a deployment and verify it is successful.
```
Comment (monopole) [inline examples/alphaTestExamples/helloapp.md]: `kapply`  `MYGOBIN`

File: examples/alphaTestExamples/helloapp.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+metadata:
+  name: the-map2
+data:
+  altGreeting: "Good Evening!"
+  enableRisky: "false"
+EOF
+
+rm $BASE/configMap.yaml
+
+kapply apply -f $BASE;
```
Comment (monopole) [inline examples/alphaTestExamples/helloapp.md]: `-f` is going away.

Also, can we add some confirmation here that the kind cluster deleted the right thing?

**Comment by mortent** [comment]
/lgtm
Looks good to me, but we should address the comments from @monopole

**Pull request merged**